fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/api/tools/ToolAttributeHolder.java

```diff
@@ -44,9 +46,18 @@
 	 * If you do not want to append on a given item stack, add an NBT tag with the key "fabric_IgnoreDynamicModifiers".
 	 * @param slot The equipment slot this item is equipped in.
 	 * @param stack The stack that's equipped.
-	 * @return The dynamic modifiers to add on top of other modifiers on this stack.
+	 * @return The dynamic modifiers to add on top of other modifiers on this stack. If none, return {@link #EMPTY}.
 	 */
 	default Multimap<String, EntityAttributeModifier> getDynamicModifiers(EquipmentSlot slot, ItemStack stack) {
-		return HashMultimap.create();
+		return EMPTY;
+	}
+
+	/**
+	 * @param slot The equipment slot this item is equipped in.
+	 * @param stack The stack that's equipped.
+	 * @return Whether the tool's dynamic modifiers should be appended.
+	 */
+	default boolean useDynamicModifiers(EquipmentSlot slot, ItemStack stack) {
+		return !stack.hasTag() || !stack.getTag().containsKey("fabric_IgnoreDynamicModifiers");
```

Comment:
I still think this solution is worth discussing. Why should fabric API have this kind of logic?

Reply:
That's a fair point. I put it in because we're technically disrupting vanilla behavior with the dynamic modifiers, and I wanted a way to turn that off. Using NBT tags is what vanilla *always* does in this situation, so I added a tag check in.